proto/substrait/algebra.proto

```diff
@@ -233,6 +237,10 @@ message SortRel {
   RelCommon common = 1;
   Rel input = 2;
   repeated SortField sorts = 3;
+
+  // True if this node only sorts a portion of the final result. If it is true,
+  // a local merge or merge exchange is required to merge the sorted runs.
```

Comment:
This needs more explanation.  Also, I don't like the term "partial".  A "partial sort", as defined by [Wikipedia](https://en.wikipedia.org/wiki/Partial_sorting) is what might be called "top N".  It sorts only part of a range.  This is also what we get from the [C++ standard library](https://en.cppreference.com/w/cpp/algorithm/partial_sort).  These are the top 2 google results for "partial sort".

To be honest, I'm not really sure what is intended here.  I assume this is based on velox.  However, the documentation on this parameter is rather sparse.  From scanning the code the only place I see this get used is to determine if the driver can use more than 1 thread.  The orderby operator itself does not actually appear to use this flag.

My educated guess is that this is related to a distributed plan (in Velox I think a "multi-threaded" plan is treated the same as "locally distributed").  So if this is a partial sort and there are multiple splits then each split can be given its own thread and sorted independently.  If it is not a partial sort and there are multiple splits then the splits must be processed one at a time and the data from all the splits needs to be accumulated and sorted.

We have had very little discussion on how Substrait will look in a distributed environment but I think it is something we very much want to solve.  However, in my mind, the "partialness" of a sort would not be something that had to be expressed in the plan.  If, for example, a plan is to be distributed to 10 workers, then I think you would create a unique substrait plan for each of the workers (and each worker would see the sort as a non-partial sort because the worker only knows about the data it is processing and there is no concept of a partial sort).

The velox approach seems more complicated, but I also don't want to reject it.  @mbasmanova can you verify some of my guessing above?  Do you have any thoughts or opinions on how we might want to tackle distributed plans in Substrait?

Reply:
To clarify, I'm also super supportive on making sure Substrait supports Velox well, just want to make sure we're not mapping things too literally.